wis2box/oregon/main.py

```diff
@@ -0,0 +1,298 @@
+###############################################################################
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+###############################################################################
+
+import json
+import requests
+import io
+from typing import Tuple, TypedDict, Optional, List
+import click
+import logging
+from wis2box import cli_helpers, data
+from wis2box.api import remove_collection, setup_collection, upsert_collection_item
+from urllib.parse import parse_qs, urlencode, urlparse
+from requests import Session
+
+from wis2box.api.backend import load_backend
+from wis2box.api.backend.sensorthings import SensorthingsBackend
+from wis2box.env import API_BACKEND_URL, STORAGE_INCOMING
+from wis2box.oregon.cache import ShelveCache
+from wis2box.oregon.lib import OregonHttpClient, download_oregon_tsv, generate_phenomenon_time, parse_date, parse_oregon_tsv
+from wis2box.oregon.types import ALL_RELEVANT_STATIONS, POTENTIAL_DATASTREAMS, Attributes, OregonHttpResponse, StationData, Datastream
+from wis2box.storage import put_data
+
+LOGGER = logging.getLogger(__name__)
+
+THINGS_COLLECTION = "Things"
+
+
+class OregonStaRequestBuilder():
+
+    relevant_stations: List[int]
+
+    def __init__(self, relevant_stations: List[int]) -> None:
+        self.relevant_stations = relevant_stations
```

Comment:
Let me know if you like the builder pattern or think there is a better way to structure this code. I also found this in RISE. Multi step ETL with nested json quickly gets kinda messy. That being said there isn't a great way to enforce order in instance methods so maybe there is a better way